Twilio/Security/RequestValidator.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Twilio.Security
+{
+    public class RequestValidator
+    {
+        private readonly HMACSHA1 _hmac;
+
+        public RequestValidator(string secret)
+        {
+            _hmac = new HMACSHA1(Encoding.UTF8.GetBytes(secret));
+        }
+
+        public bool Validate(string url, Dictionary<string, string> parameters, string expected)
```

Comment:
Think we should add an overloaded version of Validate that takes a `NameValueCollection` instead of `IDictionary<string, string>`.

It can just convert the `NameValueCollection` to a `Dictionary<string, string>` and then call that version of `Validate`.

Reason being, the HttpRequestBase.QueryString and HttpRequestBase.Form properties are a NameValueCollection, so it would make it easy to just pass it in without having to write your own dictionary conversion code.

NameValueCollection to Dictionary conversion:
http://stackoverflow.com/questions/16549758/copy-key-values-from-namevaluecollection-to-generic-dictionary